protocols/virtualdefund/virtualdefund.go

```diff
@@ -217,6 +248,71 @@ func (o Objective) fullySigned() bool {
 	return true
 }
 
+// isAlice returns true if the receiver represents participant 0 in the virtualdefund protocol
+func (o Objective) isAlice() bool {
+	return o.MyRole == 0
+}
+
+// isBob returns true if the receiver represents participant 2 in the virtualdefund protocol
+func (o Objective) isBob() bool {
+	return o.MyRole == 2
+}
+
+// ledgerProposal generates a ledger proposal to remove the guarantee for V for ledger
+func (o Objective) ledgerProposal(ledger *consensus_channel.ConsensusChannel) consensus_channel.Proposal {
+	left := o.finalOutcome().Allocations[0].Amount
+	right := o.finalOutcome().Allocations[1].Amount
+
+	return consensus_channel.NewRemoveProposal(ledger.Id, FinalTurnNum, o.VId(), left, right)
+}
+
+// defundLedger updates the ledger channel to remove the guarantee that funds V.
+func (o *Objective) defundLedger(ledger *consensus_channel.ConsensusChannel, sk *[]byte) (protocols.SideEffects, error) {
+
+	var sideEffects protocols.SideEffects
+
+	proposed := ledger.RemoveProposedFor(o.VId())
+
+	if ledger.IsLeader() {
+		if proposed { // If we've already proposed a remove proposal we can return
+			return protocols.SideEffects{}, nil
+		}
+
+		signedProposal, err := ledger.Propose(o.ledgerProposal(ledger), *sk)
+		if err != nil {
+			return protocols.SideEffects{}, fmt.Errorf("error proposing ledger update: %w", err)
+		}
+
+		message := o.createSignedProposalMessage(signedProposal, ledger)
+		sideEffects.MessagesToSend = append(sideEffects.MessagesToSend, message)
+
+	} else {
+
+		if proposed {
+			sp, err := ledger.SignNextProposal(o.ledgerProposal(ledger), *sk)
+
+			if err != nil {
+				return protocols.SideEffects{}, fmt.Errorf("could not sign proposal: %w", err)
+			}
+
+			message := o.createSignedProposalMessage(sp, ledger)
+			sideEffects.MessagesToSend = append(sideEffects.MessagesToSend, message)
+
+		} else {
+
+		}
```

Comment:
You definitely don't need this else block :-)

Reply:
Removed in https://github.com/statechannels/go-nitro/pull/540/commits/1e7a20f0d30ba31c84a43da41854c91156d2e381